src/main/java/com/strangeone101/holoitemsapi/tracking/TrackingManager.java

```diff
@@ -0,0 +1,155 @@
+package com.strangeone101.holoitemsapi.tracking;
+
+import com.google.gson.stream.JsonToken;
+import com.strangeone101.holoitemsapi.block.Placeable;
+import com.strangeone101.holoitemsapi.item.CustomItemManager;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import org.bukkit.Bukkit;
+import org.bukkit.Chunk;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import xyz.holocons.mc.holoitemsrevamp.HoloItemsRevamp;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Big shoutout to Flo0 (a.k.a 7Smiley7 on Spigot) for giving permission on copying the concept of tracking blocks.
+ * @see <a href="https://www.spigotmc.org/threads/tracking-blocks-that-were-placed-by-players.500216/">Resource thread</a>
+ */
+public class TrackingManager {
+
+    public static final String FILENAME = "tracker.json";
+
+    private final Map<UUID, TrackedWorld> trackedWorldMap;
+    private final HoloItemsRevamp plugin;
+
+    public TrackingManager(HoloItemsRevamp plugin) {
+        this.trackedWorldMap = new Object2ObjectOpenHashMap<>();
+        this.plugin = plugin;
+    }
+
+    public void loadTrackedWorlds() {
+        try {
+            final var file = new File(plugin.getDataFolder(), FILENAME);
+
+            if (!file.exists())
+                return;
+
+            if (!trackedWorldMap.isEmpty())
+                throw new IllegalStateException("Tracker already has data!");
+
+            final var reader = new GsonReader(file);
+
+            if (reader.peek() == JsonToken.NULL) {
+                reader.close();
+                return;
+            }
+
+            reader.beginObject();
+            while (reader.hasNext()) {
+                var uidString = reader.nextName();
+                UUID worldUUID;
+                try {
+                    worldUUID = UUID.fromString(uidString);
+                    if (Bukkit.getWorlds().stream().map(World::getUID).noneMatch((uniqueId) -> uniqueId.equals(worldUUID)))
+                        continue; // ignore worlds that aren't loaded.
```

Comment:
Instead of outright skipping over worlds that aren't loaded, we should mark it invalid somehow and still save it when the server shuts down. In case we happen to have some instance where we didn't load a world one time, but we do again later, we won't just lose the data.

Reply:
Hmm. It would only increase in size if players placed custom items in the resource worlds. Though it'll probably be okay to just ignore what I suggested there.

Reply:
I think to avoid losing data, we could add another method that writes to a separate json file containing the invalid data. That way if we ever need the data back, we can put it on the tracker.json. And we can just clean it up every week/2